Uncaught exception: Error: Cannot create a string longer than 0x1fffffe8 characters

Type: Bug Report

## Your Environment

- Version used: 
  - v4.6.1
  - 9eeff18
- Environment name and version:
  - Node.js version `v16.7.0`
  - npm version `7.21.1`
  - Web browser irrelevant
- Operating system and version:  Manjaro Linux aarch64
- Torrent client and version: rtorrent 0.9.8

## Summary

Starting flood by calling `npm run start -- --host 0.0.0.0 --port 3000`  fails with the following exception:
`Uncaught exception: Error: Cannot create a string longer than 0x1fffffe8 characters`.

Full output is:
```
➜  npm run start -- --host 0.0.0.0 --port 3000

> flood@4.6.1 start
> node --use_strict dist/index.js "--host" "0.0.0.0" "--port" "3000"

Flood server 4.6.1 starting on http://0.0.0.0:3000

(node:297655) [DEP0111] DeprecationWarning: Access to process.binding('http_parser') is deprecated.
(Use 'node --trace-deprecation ...' to show where the warning was created)
FATAL internal error. Please open an issue.
Uncaught exception: Error: Cannot create a string longer than 0x1fffffe8 characters
```

## Expected Behavior

flood starts succesfully

## Steps to Reproduce

1.  Download flood source tarball (tried it with v4.6.1 and 4.5.4)
2.  `npm install`
3.  `npm run build`
4.  `npm run start -- --host 0.0.0.0 --port 3000`

## Context

I was running flood successfully before. After upgrading to v4.6.1 flood fails to start. Trying to build and use a previously working version again fails with the same exception.

Seeing the same issue. Using `jesec/flood:master-dbg` on Docker 20.10.8.

```
> flood@0.0.0-master.3563344 start:development:server
> NODE_ENV=development TS_NODE_PROJECT=server/tsconfig.json ts-node-dev --transpile-only -r tsconfig-paths/register server/bin/start.ts "--host=0.0.0.0" "--port" "3000" "--allowedpath" "/data"
[INFO] 05:51:04 ts-node-dev ver. 1.1.8 (using ts-node ver. 9.1.1, typescript ver. 4.4.3)
Flood server 0.0.0-master.3563344 starting on http://0.0.0.0:3000
(node:24) [DEP0111] DeprecationWarning: Access to process.binding('http_parser') is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
Error: Cannot create a string longer than 0x1fffffe8 characters
    at Object.slice (node:buffer:593:37)
    at Buffer.toString (node:buffer:811:14)
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:63:23)
[ERROR] 05:51:07 Error: Cannot create a string longer than 0x1fffffe8 characters
```

--- a/server/models/Users.ts
+++ b/server/models/Users.ts
@@ -31,6 +31,7 @@ class Users {
     });
 
     db.ensureIndex({fieldName: 'username', unique: true});
+    db.setAutocompactionInterval(config.dbCleanInterval);
 
     return db;
   })();

--- a/server/services/feedService.ts
+++ b/server/services/feedService.ts
@@ -21,6 +21,7 @@ class FeedService extends BaseService<Record<string, never>> {
 
   constructor(...args: ConstructorParameters<typeof BaseService>) {
     super(...args);
+    this.db.setAutocompactionInterval(config.dbCleanInterval);
 
     this.onServicesUpdated = async () => {
       // Execute once only.

--- a/server/services/notificationService.ts
+++ b/server/services/notificationService.ts
@@ -26,6 +26,7 @@ class NotificationService extends BaseService<NotificationServiceEvents> {
 
   constructor(...args: ConstructorParameters<typeof BaseService>) {
     super(...args);
+    this.db.setAutocompactionInterval(config.dbCleanInterval);
 
     (async () => {
       const notifications = await this.db.findAsync<Notification>({}).catch(() => undefined);

--- a/server/services/settingService.ts
+++ b/server/services/settingService.ts
@@ -21,6 +21,11 @@ class SettingService extends BaseService<SettingServiceEvents> {
     filename: path.join(config.dbPath, this.user._id, 'settings', 'settings.db'),
   });
 
+  constructor(...args: ConstructorParameters<typeof BaseService>) {
+    super(...args);
+    this.db.setAutocompactionInterval(config.dbCleanInterval);
+  }
+
   async destroy(drop: boolean) {
     if (drop) {
       await this.db.dropDatabaseAsync();

